model_api/python/model_api/models/detection_model.py

```diff
@@ -92,63 +91,35 @@ def _resize_detections(self, detections: list[Detection], meta):
                 pad_left = (self.w - round(input_img_widht / inverted_scale_x)) // 2
                 pad_top = (self.h - round(input_img_height / inverted_scale_y)) // 2
 
-        def _clamp_and_round(val, min_value, max_value):
-            return round(max(min_value, min(max_value, val)))
+        boxes = detection_result.bboxes
+        boxes[:, 0::2] = (boxes[:, 0::2] * self.w - pad_left) * inverted_scale_x
+        boxes[:, 1::2] = (boxes[:, 1::2] * self.h - pad_top) * inverted_scale_y
+        np.round(boxes, out=boxes)
+        boxes[:, 0::2] = np.clip(boxes[:, 0::2], 0, input_img_widht)
+        boxes[:, 1::2] = np.clip(boxes[:, 1::2], 0, input_img_height)
+        detection_result.bboxes = boxes.astype(np.int32)
```

Comment:
Just in case, round and as_type could give different results, affecting the accuracy test

Reply:
Previously, we had
```python
def _clamp_and_round(val, min_value, max_value):
        return round(max(min_value, min(max_value, val)))
```
which is not equivalent to astype, also in c++ we have clamp + round combination:
```c++
clamp(round((detections[i * numAndStep.objectSize + 3] * netInputWidth - padLeft) * invertedScaleX),
                      0.f,
                      floatInputImgWidth);
```

Reply:
I see `np.round(boxes, out=boxes)` a line before, so it's fine